generated/nidaqmx/task.py

```diff
@@ -649,7 +662,22 @@ def register_done_event(self, callback_method):
                 Passing None for this parameter unregisters the event callback
                 function.
         """
-        self._interpreter.register_done_event(self._handle, 0, callback_method, None)
+        if callback_method is not None:
+            # Pass the user a Task object instead of a task handle
+            def call_done_event_callback(task_handle, status, callback_data):
+                return callback_method(self, status, callback_data)
```

Comment:
It would also be good to remove the callback_data parameter and the need to return an int.

Reply:
I prototyped more signature changes, with a `use_new_signature=False` parameter to opt in